LeaderboardBackend/Services/Impl/LeaderboardService.cs

```diff
@@ -27,9 +30,26 @@ public async Task<List<Leaderboard>> GetLeaderboards(long[]? ids = null)
         }
     }
 
-    public async Task CreateLeaderboard(Leaderboard leaderboard)
+    public async Task<CreateLeaderboardResult> CreateLeaderboard(CreateLeaderboardRequest request)
     {
-        applicationContext.Leaderboards.Add(leaderboard);
-        await applicationContext.SaveChangesAsync();
+        Leaderboard lb = new()
+        {
+            Name = request.Name,
+            Slug = request.Slug,
+            Info = request.Info
+        };
+
+        applicationContext.Leaderboards.Add(lb);
+
+        try
+        {
+            await applicationContext.SaveChangesAsync();
+        }
+        catch (DbUpdateException e) when (e.InnerException is PostgresException { SqlState: PostgresErrorCodes.UniqueViolation })
```

Comment:
What happens with any other exceptions thrown from the DB?

Reply:
500 error. In theory that really shouldn't happen.